lib.php

```diff
@@ -164,66 +164,69 @@ function mass_enroll($cir, $course, $context, $data) {
                 $enrollablecount++;
             }
 
-            if (empty($fields[1])) {
-                $group = null;
-            } else {
-                $group = str_replace('"', '', trim($fields[1]));
-            }
-            // 2nd column?
-            if (empty($group)) {
-                $result .= "\n";
-                continue; // No group for this one.
-            }
+            // Iterate over the rest of the columns to process as group names.
+            for ($colnum = 1; $colnum < count($fields); $colnum++) {
```

Comment:
FYI this is all I added to loop over each group, the rest of the diff has just been indented

Reply:
Im wondering if we should add a little more safety here, could there be a case where this overrides another column? Im guessing that the group column is generally the LAST column in the CSV?

Reply:
Previously it was only 1 column (user identifier) + optional 2nd column for groups. 

Now it just loops over every column excluding the first it to read the rest of the columns for group names.

There's no options for other columns so I don't think it's necessary to put more safety here